### ISSUE TITLE
DB2: Support parsing special registers e.g. `CURRENT DATE`

### ISSUE THREAD

**Issue by keraion** [sqlfluff/sqlfluff#5611]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing fails for special registers such as `CURRENT DATE` or `CURRENT TIMESTAMP`.

### Expected Behaviour

These should parse successfully.

### Observed Behaviour

```sh
echo 'VALUES CURRENT DATE;' | sqlfluff parse --dialect db2 -
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        values_clause:
[L:  1, P:  1]      |            keyword:                                          'VALUES'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            expression:
[L:  1, P:  8]      |                column_reference:
[L:  1, P:  8]      |                    naked_identifier:                         'CURRENT'
[L:  1, P: 15]      |    whitespace:                                               ' '
[L:  1, P: 16]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  1, P: 16]      |        word:                                                 'DATE'
[L:  1, P: 20]      |        semicolon:                                            ';'
[L:  1, P: 21]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:  16 |  PRS | Line 1, Position 16: Found unparsable section: 'DATE;'
WARNING: Parsing errors found and dialect is set to 'db2'. Have you configured your dialect correctly?
```

### How to reproduce

See observed behavior.

### Dialect

DB2

### Version

sqlfluff, version 3.0.0a5

### Configuration

defaults

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
